lib/backend/model/workloadendpoint.go

```diff
@@ -54,22 +54,28 @@ func (key WorkloadEndpointKey) defaultPath() (string, error) {
 		key.Hostname, key.OrchestratorID, key.WorkloadID, key.EndpointID), nil
 }
 
-func (key WorkloadEndpointKey) defaultDeletePath() (string, error) {
+func (key WorkloadEndpointKey) defaultDeletePaths() (string, []string, error) {
 	if key.Hostname == "" {
-		return "", errors.ErrorInsufficientIdentifiers{Name: "hostname"}
+		return "", nil, errors.ErrorInsufficientIdentifiers{Name: "hostname"}
 	}
 	if key.OrchestratorID == "" {
-		return "", errors.ErrorInsufficientIdentifiers{Name: "orchestrator"}
+		return "", nil, errors.ErrorInsufficientIdentifiers{Name: "orchestrator"}
 	}
 	if key.WorkloadID == "" {
-		return "", errors.ErrorInsufficientIdentifiers{Name: "workload"}
+		return "", nil, errors.ErrorInsufficientIdentifiers{Name: "workload"}
 	}
 	if key.EndpointID == "" {
-		return fmt.Sprintf("/calico/v1/host/%s/workload/%s/%s/",
-			key.Hostname, key.OrchestratorID, key.WorkloadID), nil
+		return "", nil, errors.ErrorInsufficientIdentifiers{Name: "endpoint"}
 	}
-	return fmt.Sprintf("/calico/v1/host/%s/workload/%s/%s/endpoint/%s",
-		key.Hostname, key.OrchestratorID, key.WorkloadID, key.EndpointID), nil
+
+	// A workload endpoint may auto create two additional entries that need to be
+	// deleted.  This is actually dependent on the datastore, but we return these
+	// as part of the default paths.
+	workload := fmt.Sprintf("/calico/v1/host/%s/workload/%s/%s",
+		key.Hostname, key.OrchestratorID, key.WorkloadID)
+	endpoints := workload + "/endpoint"
+	endpoint := endpoints + key.EndpointID
+	return endpoint, []string{endpoints, workload}, nil
```

Comment:
OK, so the returned paths need to be ordered such that while iterating through, each path is always empty when being deleted.

Making a note to myself to make sure that this is documented somewhere :)